server/core.go

```diff
@@ -0,0 +1,69 @@
+package server
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	logging "github.com/ipfs/go-log"
+
+	"github.com/valyala/fasthttp"
+)
+
+var log = logging.Logger("core")
+
+func init() {
+	logging.SetupLogging()
+}
+
+type gateway struct {
+	server  fasthttp.Server
+	address string
+}
+
+func NewGateway(opts ...GatewayOption) *gateway {
```

Comment:
I think we should pass DFMS API as a parameter here.

DFMS API is an interface group that has several implementations and there are cases where the gateway is used within the same process as well, while current behavior forces using only HTTP API and creates it himself.

Reply:
Imagine everything as a reusable library and the first user of the gateway lib is the main func. Gateway might be also imported by other current and future modules in our ecosystem. I am considering to make a new app in dfms repo(cmd folder) called `gate` to simplify gateway builds as everything is configured for dfms repo already. That app(cause it will have main) should automatically detect dfms daemon on start and connect to it passing the HTTP API to gateway or to be configured to work with some remote node.

Reply:
The main in this repo might be still saved as a usage example, but the actual application we will use for as dfms gateway will be in its repo for simplicity.

Reply:
> I think we should pass DFMS API as a parameter here.
> 
> DFMS API is an interface group that has several implementations and there are cases where the gateway is used within the same process as well, while current behavior forces using only HTTP API and creates it himself.

Yeah, I'd thought in that way, but then made the wrong decision :D